addons/account/models/account_move.py

```diff
@@ -826,7 +826,15 @@ def _compute_tax_audit(self):
             currency = record.company_id.currency_id
             audit_str = ''
             for tag in record.tag_ids:
-                tag_amount = (tag.tax_negate and -1 or 1) * (record.journal_id.type == 'sale' and -1 or 1) * (record.invoice_id.type in ('in_refund', 'out_refund') and -1 or 1) * record.balance
+
+                # In case of cash basis entries, we need to consider the original invoice, not the current move
+                if record.move_id.tax_cash_basis_rec_id:
+                    reconciled_amls = record.move_id.tax_cash_basis_rec_id.debit_move_id + record.move_id.tax_cash_basis_rec_id.credit_move_id
+                    invoice_aml = reconciled_amls.filtered(lambda x: x.journal_id.type != 'bank') # To exclude the payment
```

Comment:
why not `lambda x: x.journal_id.type in ('sale', 'purchase')` then?

Reply:
Because that's what we did here https://github.com/odoo/enterprise/commit/7d17dad678361f161c259eddd45d9b5186941bd0#diff-a9e136ddd42a769ed7a4075978e3e28eR323 But it's all the same, actually. I just think both should be consistent.

Reply:
@smetl Nervermind. I found one more issue, so I'm gonna change that anyway. Plus, if they reconcile the invoice with a misc entry, I suppose we expect CABA to work as well, so let's do as you suggest.